### ISSUE TITLE
Determine correct order of ease factor and interval updates

### ISSUE THREAD

**Issue by joshdavham** [open-spaced-repetition/sm-2#2]
When a card is correctly recalled, it's not clear whether the ease factor or the interval should be updated first.

If we reference Piotr Wozniak's [article](https://super-memory.com/english/ol/sm2.htm) and [source code](https://super-memory.com/english/ol/sm2source.htm) on super-memory.com, the next interval is calculated first, then the ease factor is updated after. The wikipedia [page](https://en.wikipedia.org/wiki/SuperMemo#Algorithms) for SuperMemo also leads to believe this is the correct way.

Given the above, this is currently how the `sm-2` package is implemented.

However, I found this [page](https://supermemo.guru/wiki/SuperMemo_1.0_for_DOS_(1987)#Algorithm_SM-2) today from supermemo.guru which states
> After each repetition, before computing the new interval, modify the E-Factor of the recently repeated item

This makes more sense to me intuitively as you're updating how easy the card is before determining the next interval. However, it should also be noted that this section of the wiki has an [archive warning](https://supermemo.guru/wiki/Why_use_literal_archives%3F) which states that
> Archive materials are listed with minimum editorial intervention.

and

> Literal use of the archive materials may mean that some erroneous claims may be included in the text

...meaning that that article could be outdated and incorrect.

So basically, I'm not sure which is correct.

If anyone has the answer, I'd love to hear it!

**Comment by Expertium** [comment]
I think in Anki it's ease first
